### PR TITLE
DisposableChannel#address returns SocketAddress

### PR DESCRIPTION
**Pull request by violetagg**
This is related to Unix Domain Sockets support.

### CODE REVIEW

File: src/main/java/reactor/netty/DisposableServer.java [hunk 1]
```diff
@@ -17,33 +17,55 @@
...
+		return channel().localAddress();
+	}
+
 	/**
 	 * Returns the server's host string. That is, the hostname or in case the server was bound
 	 * to a literal IP address, the IP string representation (rather than performing a reverse-DNS
 	 * lookup).
 	 *
 	 * @return the host string, without reverse DNS lookup
+	 * @throws UnsupportedOperationException when Unix Domain Sockets
```
Comment (simonbasle) [inline src/main/java/reactor/netty/DisposableServer.java:42]: the default implementation returns `""` while the javadoc says it throws (I prefer the former)

File: src/main/java/reactor/netty/DisposableServer.java [hunk 2]
```diff
@@ -17,33 +17,55 @@
...
-		return address().getHostString();
+		return "";
+	}
+
+	/**
+	 * Returns the server's path to the domain socket.
+	 * Returns {@code null} in case it is not Unix Domain Sockets.
+	 *
+	 * @return the path to the domain socket
+	 * @throws UnsupportedOperationException when it is not Unix Domain Sockets
```
Comment (simonbasle) [inline src/main/java/reactor/netty/DisposableServer.java:55]: the default implementation returns `""` while the javadoc says it throws (I prefer the former)

File: src/main/java/reactor/netty/DisposableServer.java [hunk 3]
```diff
@@ -17,33 +17,55 @@
...
+	 */
+	default String path(){
+		return "";
 	}
 
 	/**
 	 * Returns this server's port.
+	 *
 	 * @return The port the server is bound to.
+	 * @throws UnsupportedOperationException when Unix Domain Sockets
```
Comment (simonbasle) [inline src/main/java/reactor/netty/DisposableServer.java:66]: the default implementation returns `-1` while the javadoc says it throws (I prefer the former)

File: src/test/java/reactor/netty/tcp/BlockingConnectionTest.java [hunk 4]
```diff
@@ -91,14 +101,15 @@ public void simpleServerFromAsyncServer() throws InterruptedException {
 				         .wiretap(true)
 				         .bindNow();
 
-		System.out.println(simpleServer.address().getHostString());
-		System.out.println(simpleServer.address().getPort());
+		InetSocketAddress address = (InetSocketAddress) simpleServer.address();
+		System.out.println(address.getHostString());
```
Comment (simonbasle) [inline src/test/java/reactor/netty/tcp/BlockingConnectionTest.java:105]: (just a note to remove the `System.out`)

File: src/main/java/reactor/netty/DisposableServer.java (hunk 1 above)
Comment (violetagg) [reply src/main/java/reactor/netty/DisposableServer.java:42]: done

File: src/main/java/reactor/netty/DisposableServer.java (hunk 2 above)
Comment (violetagg) [reply src/main/java/reactor/netty/DisposableServer.java:55]: done

File: src/main/java/reactor/netty/DisposableServer.java (hunk 3 above)
Comment (violetagg) [reply src/main/java/reactor/netty/DisposableServer.java:66]: done

File: src/test/java/reactor/netty/tcp/BlockingConnectionTest.java (hunk 4 above)
Comment (violetagg) [reply src/test/java/reactor/netty/tcp/BlockingConnectionTest.java]: removed

**Pull request closed without merging**